Packet++/src/TcpReassembly.cpp

```diff
@@ -23,13 +23,44 @@
 namespace pcpp
 {
 
-	static timeval timespecToTimeval(const timespec& in)
+	static timeval timePointToTimeval(const std::chrono::time_point<std::chrono::high_resolution_clock>& in)
 	{
-		timeval out;
-		TIMESPEC_TO_TIMEVAL(&out, &in);
+		auto duration = in.time_since_epoch();
+
+		auto seconds = std::chrono::duration_cast<std::chrono::seconds>(duration);
+		auto nanoseconds = std::chrono::duration_cast<std::chrono::nanoseconds>(duration) % 1000000000;
```

Comment:
nit: Could possibly be done without the magic number by casting the seconds to nanosecond precision and substracting that from the main duration timepoint to get the remaining nanoseconds, but I guess this works too.

Reply:
Ture, I think Dimi's method will be more efficient.

Reply:
> Ture, I think Dimi's method will be more efficient.

Looked through it in [godbolt.org](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxCCSAJykAA6oCoRODB7evnrJqY4CQSHhLFEx8baY9vkMQgRMxASZPn5cFVXptfUEhWGR0bEJCnUNTdmtQ109xaUDAJS2qF7EyOwc5gDMwcjeWADUJutuyAjEAqgH2CYaAIJXt2abVFhUuwD6ALJCbq8AatgASnchsQvA5do42AA3MR3EwAdisN12yN2Q3QIBAXmCBHWZleBHBkNeSlM60R1xRqII6Mx2Nx%2BMJry8JIO5PhABFWbCHq48FRYTcIZhobRwXg2MpUNiACqoaXi4Via4QNAMIZUmnHU4MVAYoWvZLYg5uNEYrVnDEIPDABCvYiYBSeLzVV7bVDIADWF3MADZdsFZrCEXdKczgsBdsw2ApEkwVhqzSczlybgK4kxnahduglkxqvt1uz/QwAHT61IMFavTDJY4QQNktMZoiozCq9AKAtFnPEPPpV1MIbG00oJM6jEkgQdi4QHt9gQNtlIlHNrPMHWThgdrvZ3Mu0RDw4j83jkDrlJtqcKGdz6qzXYAWl3vf3g4Iw%2Bpie1urPhgv7evdZsAgTcO0XENKRRCDkSFEVdkWd9G1TDQ4gQssiRJHdQIUEs0C8QR6xTW4ULQggiWZNsd3PbDcMWAj7wAel2LgNFYojKXtAglgYeDnSIjl%2BMFBU4KFSUZTlYSxAsFUBHVY8xx/fVDUED9NQUy1rVte1HW8fdaHdL0gN9YtA1TYNl2RMMjEjVgHVjeN5O/QTiPTTNn3nHiDiLYIywVYktkwatawQQikJc1dWwAndb37A9EJNT9R2/CdLy3QDgJihdnKuVyW2o1Lty89zX0PBK1OS38NwK9LZz3dJ73MABWZjWNatr2Mg5FoLFKExF4xCl1Q510OJSiipovD6I65FSPIzCivygDaPwgh612JiWLYsKOMwLjiB4hD%2BLhTkwrubFdhYJhgkIsylwpFc3NUHdRKlQRZXlXraGVRyLRAK0bTtB0nT0gyMR1AB3etwIs3YItULhnoVMS3okz7pJ%2B09/q0oHdNi/TPTB1BIdmaHiPu5FOO43YNCOzkbg4eZaE4RreD8bheF1Dg3GsaxUUWZZMH2B4eFIAhNAZ%2BYEEwJgsBietSA9EBGo0EtGrMMwuDiXEuCkAAOXWzB9fROEkVnxdITneAUEANFF8X5jgWAYEQFBUBYRI6GichKDQd3PZibZDGATWWL4OgCGia2IAic2ImCeoAE9OBFuPmGIBOAHkIm0NsxfZ0hfbYQQM4YWgk44LRSCwCIvGANwxFoa386wS6jHECveHwe0HDwSEHXNzBVDbZ1VhF7FKnN2g8AiXt048LBzYIYhxWT3g%2B%2BICIL3ZTBW%2BAKejHtvgDGABQfjwTBwYzxJGFXmRBBEMR2CkO/5CUNRzd0VoDAP0xLGsfRp7W1gFGdgVAWAEFIOvEAS98IehJhbRI1Qm4PgAOoN0fGiLyv8rCWC4HCR8GczAc3XsvLAQD5Z2FzukFwW5RgtFIIEYIvQSj9FaLkNIAg6E5BSBwhgUw%2BgxHGJUKhAhOgjCdGMNoIiajDG6Ew6YrDbCyK4eMWR/CWGCPmI6JYKwJCM2ZmbDuFtOC7FULrH0D4fSSF2IHaymsSxcBLBoXYEBcCEBIELdYXBZi8DzloeBUsZb9HlorZWxsOCm1ICwEAPoVY%2BjiBoSQwsfRmEarrdYatSBs0rpbWwNs7Ydwds7CASAEKJGdN7CAmB8BEFIXofg99RDiGfg01%2BKh1BGM/qQcGvZEir30RwFmWTzacwzs6cpBJUAvDMRYqxNjv4RnsY45xEAPB%2B3oMQTx3jfH20ltLWWlABmROiesOIJZ1goUanCH06S4SSC4FY3WwyjG5OtrbPxEsFYDBLJIOEcJdYaB9P83WjUdbXPCesQxOTOA7MKQMohUSJCsWedCjgsL/HzHXqkZwkggA%3D%3D%3D). It actually does not matter under optimizations. Both compile down to mostly the same assembly.

Reply:
Amazing compiler

Reply:
I agree! Fixed in https://github.com/seladb/PcapPlusPlus/pull/1591/commits/91f6b390a2c23412c301f6183ba634491106feec